src/components/references/ReferenceSearchSection.tsx

```diff
@@ -0,0 +1,130 @@
+/*!
+ * © 2023 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ArrowDownUp from '@manuscripts/assets/react/ArrowDownUp'
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import { BibliographyItemAttrs } from '../../lib/references'
+import {
+  BibliographyItems,
+  BibliographyItemSource,
+  CancellablePromise,
+} from './BibliographyItemSource'
+import { ReferenceSearchResults } from './ReferenceSearchResults'
+import { ReferenceSearchResultsPlaceholder } from './ReferenceSearchResultsPlaceholder'
+
+const SearchSourceLabel = styled.div`
+  margin: 0 ${(props) => props.theme.grid.unit * 4}px
+    ${(props) => props.theme.grid.unit * 2}px;
+  color: ${(props) => props.theme.colors.text.secondary};
+  cursor: pointer;
+
+  &:hover {
+    color: ${(props) => props.theme.colors.text.muted};
+  }
+`
+
+export const Arrow = styled(ArrowDownUp)`
+  margin-right: 16px;
+  margin-left: 10px;
+
+  user-select: none;
+
+  transform: rotate(180deg);
+
+  &.open {
+    transform: rotate(0deg);
+  }
+`
+
+type RunningState = {
+  type: 'running'
+  promise: CancellablePromise<BibliographyItems>
+}
+
+type CompletedState = {
+  type: 'completed'
+  items: BibliographyItemAttrs[]
+  total: number
+}
+
+type State = RunningState | CompletedState
+
+export const ReferenceSearchSection: React.FC<{
+  query: string
+  source: BibliographyItemSource
+  isSelected: (item: BibliographyItemAttrs) => boolean
+  onSelect: (item: BibliographyItemAttrs) => void
+}> = ({ query, source, isSelected, onSelect }) => {
+  const [expanded, setExpanded] = useState(true)
+  const [state, setState] = useState<State>()
```

Comment:
adding state in this component will not work with the search, any change in the props especially `source` will reset state to undefined. if we move the state to parent that would help us to fix the issue of search

Reply:
Not sure what you mean. This code has been there for a while, and is currently working fine in stag.

Reply:
Maybe that issue was just for me locally!

Reply:
If you were trying it locally, there was indeed a problem with search, but it has to do with `crossref.ts`. I just pushed a fix for that.

Reply:
thanks, it fixed the issue